cli/src/main/java/org/wildfly/glow/jbang/JBangIntegration.java

```diff
@@ -0,0 +1,186 @@
+package org.wildfly.glow.jbang;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.wildfly.glow.GlowSession;
+
+public class JBangIntegration {
+
+    public static void extractJar(Path jar, Path destDir) throws IOException {
+        try (JarFile jarFile = new JarFile(jar.toFile())) {
```

Comment:
Just a general comment. I find the `FileSystem` and NIO file API to work a little better in cases like this as they throw better exceptions if something fails. I'm not saying we need to just this though.

Reply:
+1, I'll update to use more consistently the API (they are all other the places atm) :)

Reply:
Sure. It's definitely not critical especially for a first commit.

Reply:
I still did it, that makes for a clearer code :)